### PR TITLE
Add writing support for lookup type 6, formats 1 and 3

### PR DESCRIPTION
**Pull request by Connum**
## Description

This allows to write fonts containing tables of lookup type 6, in formats 1 and 3.

## Motivation and Context

Fonts containing such tables would simply fail when writing. Lookup type 6 is needed for the Chaining Contextual Substitution feature.

I originally planned to implement all three formats for this type, but this PR now only implements format 1 and 3 and will still throw a descriptive error when encountering format 2, for the following reasons:
1. The class based lookups seem to be more complex than the others, and I'm not quite sure yet whether we have everything ready in terms of data types for writing these (but I also didn't have the time to look into this further)
2. I didn't find a font file containing a table with format 6, type 2 to test this with and given the complexity, I didn't want to take this on based on the specs alone
3. I ran out of time due to work projects

**If anyone can provide a font using format 6, type 2, I would be happy to look into this and implement it, if and when my free time allows. I would also be interested in implementing all other missing types (5, 7 and 8) in order to complete the writing functionality of the library.**

## How Has This Been Tested?

I added test cases for writing both formats according to [the specs](https://docs.microsoft.com/de-de/typography/opentype/spec/gsub#lookuptype-6-chaining-contextual-substitution-subtable). I also loaded and saved several font files containing such tables, checking that they contained the corresponding tables afterwards.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue) - _fixes #309_ 
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I did `npm run test` and all tests passed green (including code styling checks).
- [X] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the **README** accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: test/tables/gsub.js [hunk 1]
```diff
@@ -363,4 +363,120 @@ describe('tables/gsub.js', function() {
             ]
         }), expectedData);
     });
+
+    //// Lookup type 4 ////////////////////////////////////////////////////////
+    // is already tested above
```
Comment (Jolg42) [inline test/tables/gsub.js:368]: That one I guess https://github.com/opentypejs/opentype.js/pull/453/files#diff-69ff5caba80b4c351f1e25cb532d195b016fbe21b912a272d1820a2bd99ea8eaR107-R130

**Review by Jolg42 — APPROVED** [review]
Well done! 💯

**Comment by calebmer** [comment]
> If anyone can provide a font using format 6, type 2, I would be happy to look into this and implement it

Inter seems to contain format 6, type 2. I ran into the error "lookup type 6 format 2 is not yet supported" when trying to modify Inter.

https://rsms.me/inter/

**Comment by spaceemotion** [comment]
@calebmer if it's of any help, I found that the 3.0 release of Inter works with this library. Seems like the format 2 thing has been introduced within that release cycle.

**Comment by calebmer** [comment]
Oh thanks! I ended up using Python's fonttools ttx for the modification I needed to make.

**Pull request merged**